haskell.nix/library/.github/workflows/check-nightly.yml

```diff
@@ -0,0 +1,39 @@
+on:
+  schedule:
+    # runs every Monday at 10:00 am UTC
+    - cron: '0 10 * * 1'
+
+jobs:
+  check-nightly:
+    # We use shared runners instead of our selfhosted ones because the stack integrates quite poorly with NixOS
+    runs-on: ubuntu-latest
+    outputs:
+      resolver: ${{ steps.nightly.outputs.resolver }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: build with nightly
+        id: nightly
+        run: |
+          # get the latest nightly resolver
+          resolver=$(curl -s -w %{redirect_url} https://www.stackage.org/nightly | sed 's:.*/::')
+          echo "resolver=$resolver" >> $GITHUB_OUTPUT
+          stack build --resolver $resolver
+
+  report-failure:
+    runs-on: ubuntu-latest
+    if: failure()
+    needs: check-nightly
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Report build failure
+        env:
+          RESOLVER: ${{ needs.check-nightly.outputs.resolver }}
+        run: |
+          # you can change 'libraries' to the channel dedicated to the library, if there is one
+          # in order for a bot to be able to post messages, it must be added to the appropriate channel
+          curl -XPOST https://slack.com/api/chat.postMessage \
```

Comment:
Maybe create a one-line script with this `curl` command and reuse it here and in `check-outdated.yml`?

Reply:
Done